test/parallel/test-stream-filter.js

```diff
@@ -87,20 +87,11 @@ const { setTimeout } = require('timers/promises');
 
 {
   // Error cases
-  assert.rejects(async () => {
-    // eslint-disable-next-line no-unused-vars
-    for await (const unused of Readable.from([1]).filter(1));
```

Comment:
To do this you probably have to make filter a non async generator as well (just removing the `async function*` bit and changing the `yield*` to `return` should work)